__tests__/commands/install/integration.js

```diff
@@ -865,6 +865,29 @@ test.concurrent('install will not overwrite files in symlinked scoped directorie
   );
 });
 
+test.concurrent('install will not overwrite linked dependencies', async (): Promise<void> => {
+  await runInstall(
+    {},
+    'install-dont-overwrite-linked',
+    async (config): Promise<void> => {
+      const dependencyPath = path.join(config.cwd, 'node_modules', 'fake-dependency');
```

Comment:
I don't understand the test.
i would expect that you would run a link command and then install should not overwrite it?

Reply:
A few tips.

0. You probably want to setup several folders for the test: a linked package.json and the linker package.json
1. here is how link command is used in tests https://github.com/yarnpkg/yarn/blob/master/__tests__/commands/link.js#L35
2. after linking you can check that link exists
3. after that run install command and check that link was not destroyed
https://github.com/yarnpkg/yarn/blob/master/__tests__/commands/install/integration.js#L93

Reply:
Thank you! Yeah I dont get this test either. I copied it from the other one that checks if scoped links are maintained, I suspect that one is faulty too. 

I'll update both with your suggestions here.